File: problems/array/1.twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var temp = []
var twoSum = function(nums, target) {
    for(i=0; i<nums.length;i++){
        if(nums[i]>target){
            continue;
        }else if(temp.length ===1 && nums[i]+nums[temp[0]] === target){
            temp.push(i)
        }else {
            temp=[i]
        }
    }
    return temp
};

File: problems/array/1.twoSum.py
"""#1 Two Sum
Pattern: hash-map complement lookup.
Time: O(n) | Space: O(n)
"""


class Solution:
    def twoSum(self, nums, target):
        seen = {}
        for index, number in enumerate(nums):
            complement = target - number
            if complement in seen:
                return [seen[complement], index]
            seen[number] = index
